### PR TITLE
Remove Unnecessary Subclasses from test_excel

### PR DESCRIPTION
**Pull request by WillAyd**
More cleanup...

### CODE REVIEW

File: pandas/tests/io/test_excel.py [hunk 1]
```diff
@@ -2065,11 +2065,10 @@ def test_path_local_path(self, merge_cells, engine, ext):
 
 
 @td.skip_if_no('openpyxl')
-@pytest.mark.parametrize("merge_cells,ext,engine", [
-    (None, '.xlsx', 'openpyxl')])
-class TestOpenpyxlTests(_WriterBase):
+@pytest.mark.parametrize("ext", ['.xlsx'])
```
Comment (WillAyd) [inline pandas/tests/io/test_excel.py]: Only one test in each of these classes was actually using the `engine` parameter so figured cleaner to just specify there

File: pandas/io/excel/_base.py [hunk 2]
```diff
@@ -591,7 +591,7 @@ class ExcelWriter(metaclass=abc.ABCMeta):
     def __new__(cls, path, engine=None, **kwargs):
         # only switch class if generic(ExcelWriter)
 
-        if issubclass(cls, ExcelWriter):
+        if cls is ExcelWriter:
```
Comment (WillAyd) [inline pandas/io/excel/_base.py]: This isn't a user-facing bug but trying to import any of the ExcelWriter subclasses would exhibit rather surprising behavior which causes a test failure when getting rid of the subclassing. For instance:

```python
>>> from pandas.io.excel._openpyxl import _OpenpyxlWriter
>>> writer = _OpenpyxlWriter('foo.xlsx')
>>> type(writer)
<class 'pandas.io.excel._xlsxwriter._XlsxWriter'>
```

From the comment directly preceding this it appears this condition was backwards, so changed as such

File: pandas/tests/io/test_excel.py (hunk 1 above)
Comment (WillAyd) [reply pandas/tests/io/test_excel.py]: `merge_cells` was also never used but rather required due to the mix of pytest / subclassing paradigms; I've done away with it entirely here

**Comment by jreback** [comment]
@WillAyd you have the other excel test PR; this lgtm. so merge if going to be a conflict (or rebase this after)

**Pull request merged**